File: Golang/DivideTwoIntegers/DivideTwoIntegers_test.go
package DivideTwoIntegers

import "testing"

func TestDivide(t *testing.T) {

	if divide(1, 1) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if divide(0, 1) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if divide(-1, -1) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if divide(intMaxValue, 1) == intMaxValue {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if divide(intMaxValue, intMinValue) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if divide(intMinValue, -1) == intMaxValue {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if divide(100, 6) == 16 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/DivideTwoIntegersTest.py
from unittest import TestCase

from DivideTwoIntegers import DivideTwoIntegers


class TestDivideTwoIntegers(TestCase):
    def test_divide(self):
        d = DivideTwoIntegers()

        self.assertTrue(d.divide(1, 1) == 1)

        self.assertTrue(d.divide(0, 1) == 0)

        self.assertTrue(d.divide(-1, -1) == 1)

        self.assertTrue(d.divide(2147483647, 1) == 2147483647)

        self.assertTrue(d.divide(2147483647, -2147483648) == 0)

        self.assertTrue(d.divide(-2147483648, -1) == 2147483647)

        self.assertTrue(d.divide(100, 6) == 16)
